pyomo/contrib/parmest/parmest.py

```diff
@@ -23,133 +42,46 @@
 )
 
 import pyomo.environ as pyo
+
 from pyomo.opt import SolverFactory
 from pyomo.environ import Block, ComponentUID
 
 import pyomo.contrib.parmest.mpi_utils as mpiu
 import pyomo.contrib.parmest.ipopt_solver_wrapper as ipopt_solver_wrapper
 import pyomo.contrib.parmest.graphics as graphics
 
-pysp, pysp_available = attempt_import('pysp')
-
-parmest_available = numpy_available & pandas_available & scipy_available & \
-                    pysp_available
+parmest_available = numpy_available & pandas_available & scipy_available
 
 inverse_reduced_hessian, inverse_reduced_hessian_available = attempt_import(
     'pyomo.contrib.interior_point.inverse_reduced_hessian')
 
-st = attempt_import('pysp.util.rapper')[0]
-scenariotree = attempt_import('pysp.scenariotree')[0]
-
 logger = logging.getLogger(__name__)
 
-__version__ = 0.1
-
-
-#=============================================
-def _object_from_string(instance, vstr):
-    """
-    Create a Pyomo object from a string; it is attached to instance
-    args:
-        instance: a concrete pyomo model
-        vstr: a particular Var or Param (e.g. "pp.Keq_a[2]")
-    output:
-        the object
-    NOTE: We need to deal with blocks
-          and with indexes that might really be strings or ints
-    """
-    # pull off the index
-    l = vstr.find('[')
-    if l == -1:
-        indexstr = None
-        basestr = vstr
+def ef_nonants(ef):
+    # Wrapper to call someone's ef_nonants
+    # (the function being called is very short, but it might be changed)
+    if use_mpisppy:
+        return sputils.ef_nonants(ef)
```

Comment:
Since the `ef_nonants` function has the same name and arguments regardless of if we're using mpi-sppy or not can we just explicitly import it above and avoid this wrapper? Is the comment implying that this is for future-proofing in case the functions diverge?

Reply:
I was not clever enough to think of using "future-proofing," but yes, I think we should keep it this way because I think it might
help future maintainers who might not be aware of the relationship and history between the stochastic programming terminology and least squares parameter estimation.